Sources/XcodeGenKit/SourceGenerator.swift

```diff
@@ -77,13 +77,19 @@ class SourceGenerator {
         localPackageGroup!.children.append(fileReference)
     }
 
-    func getAllSourceFiles(targetType: PBXProductType, sources: [TargetSource]) throws -> [SourceFile] {
-        try sources.flatMap { try getSourceFiles(targetType: targetType, targetSource: $0) }
+    /// Collects an array complete of all `SourceFile` objects that make up the target based on the provided `TargetSource` definitions.
+    ///
+    /// - Parameters:
+    ///   - targetType: The type of target that the source files should belong to.
+    ///   - sources: The array of sources defined as part of the targets spec.
+    ///   - buildPhases: A dictionary containing any build phases that should be applied to source files at specific paths in the event that the associated `TargetSource` didn't already define a `buildPhase`. Values from this dictionary are used in cases where the project generator knows more about a file than the spec/filesystem does (i.e if the file should be treated as the targets Info.plist and so on).
+    func getAllSourceFiles(targetType: PBXProductType, sources: [TargetSource], buildPhases: [Path : BuildPhaseSpec]) throws -> [SourceFile] {
```

Comment:
I'm not sure if trying to retroactively add documentation to this method myself is a great idea. I probably didn't describe it as well as I need to, but the important bit here was for me to explain what the `buildPhases` argument does.